podcasts/End of Year/StoryShareableText.swift

```diff
@@ -0,0 +1,89 @@
+import UIKit
+import PocketCastsServer
+import PocketCastsDataModel
+
+class StoryShareableText: UIActivityItemProvider {
+    private var text: String
+
+    private let hashtags = "#pocketcasts #endofyear2022"
+    private let pocketCastsUrl = "https://pca.st/"
+
+    private var shortenedURL: String?
+    private var longURL: String?
+    private var podcastListURL: String?
+
+    init(_ text: String) {
+        self.text = "\(text) \(hashtags) \(pocketCastsUrl)"
+        super.init(placeholderItem: self.text)
+    }
+
+    init(_ text: String, podcast: Podcast) {
+        self.text = "\(text) \(hashtags)"
+        super.init(placeholderItem: self.text)
+        self.longURL = podcast.shareURL
+        requestShortenedURL()
+    }
+
+    init(_ text: String, episode: Episode) {
+        self.text = "\(text) \(hashtags)"
+        super.init(placeholderItem: self.text)
+        self.longURL = episode.shareURL
+        requestShortenedURL()
+    }
+
+    init(_ text: String, podcasts: [Podcast]) {
+        self.text = "\(text) \(hashtags)"
+        super.init(placeholderItem: self.text)
+        podcastListURL = ""
+        createList(from: podcasts)
+    }
+
+    override var item: Any {
+        if let longURL {
+            return String(format: text, shortenedURL ?? longURL)
+        }
+
+        if let podcastListURL {
+            return String(format: text, podcastListURL)
+        }
+
+        return text
+    }
+
+    private func requestShortenedURL() {
+        guard let longURL, let url = URL(string: longURL) else {
+            return
+        }
+
+        let task = URLSession(configuration: .default, delegate: self, delegateQueue: nil).dataTask(with: url) { _, _, _ in }
+
+        task.resume()
+    }
+
+    private func createList(from podcasts: [Podcast]) {
+        if let shareUrl = Settings.top5PodcastsListLink {
```

Comment:
I'm still not 100% sure we'll have that on the final version of the feature but it's implemented anyway, if we don't want it we'll revert commit `de46a9b144f8eae3f4cb4f6741dabd0acb9a4df0`